Make flushing trace_log under MSan 25% faster by disabling frame pointers for demangling

azat opened the pull request:
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Refs: https://github.com/ClickHouse/ClickHouse/issues/91410

azat committed:
Remove unused tests/config/lsan_suppressions.txt

azat committed:
Make flushing trace_log under MSan faster by disabling instrumentation for __cxa_demangle

I've tested and demangling takes 50% of time, for 12K entries, 3.6s w/
demangle and 1.8-4sec w/o

Algunenano reviewed (approved):
Let's see how it goes. Do you think it'd be an option to keep a small-ish cache of symbol -> demangled symbol? I'd expect most of the symbols, specially at the top of the tree, to appear over and over

alexey-milovidov reviewed (commented):
Alternative solution is to disable demangling of trace_log under MSan.

azat committed:
Faster SYSTEM FLUSH LOGS under MSan (demangling is slow with frame pointers)

Disable frame pointers for libcxxabi to reduce demangle cost, llvm
parser has some recursion there.

This makes SYSTEM FLUSH LOGS 2x faster

azat committed:
Revert "Make flushing trace_log under MSan faster by disabling instrumentation for __cxa_demangle"

The problem was not in MSan instrumentation, but in frame pointers

This reverts commit b11cc93f8d1fb54af029ad90b2dcf899397c1aa4.

azat commented:
Turns out the problem was due to frame-pointers, confirmed manually, works 2x faster

azat commented:
>Alternative solution is to disable demangling of trace_log under MSan.

Sure, but I want to avoid difference for MSan if possible, and now, w/o frame-pointers it does not have this overhead

Algunenano reviewed (commented):
Why do we need to omit frame pointers when it's the default for the compiler and the project? Who's enabling them?
